router/router.go

```diff
@@ -331,6 +282,90 @@ func (rt *HandleT) workerProcess(worker *workerT) {
 	}
 }
 
+func (rt *HandleT) handlePrevFailedUserJob(job *jobsdb.JobT, paramaters ParametersT, userID string, worker *workerT, previousFailedJobID int64) bool {
+	// job is behind in queue of failed job from same user
+	if previousFailedJobID < job.JobID {
+		logger.Debugf("[%v Router] :: skipping processing job for userID: %v since prev failed job exists, prev id %v, current id %v", rt.destName, userID, previousFailedJobID, job.JobID)
+		resp := fmt.Sprintf(`{"blocking_id":"%v", "user_id":"%s"}`, previousFailedJobID, userID)
+		status := jobsdb.JobStatusT{
+			JobID:         job.JobID,
+			AttemptNum:    job.LastJobStatus.AttemptNum,
+			ExecTime:      time.Now(),
+			RetryTime:     time.Now(),
+			JobState:      jobsdb.Waiting.State,
+			ErrorResponse: []byte(resp), // check
+		}
+		rt.responseQ <- jobResponseT{status: &status, worker: worker, userID: userID}
+		return true
+	}
+	if previousFailedJobID != job.JobID {
+		panic(fmt.Errorf("previousFailedJobID:%d != job.JobID:%d", previousFailedJobID, job.JobID))
+	}
+	return false
+}
+
+func (rt *HandleT) handleBackoff(job *jobsdb.JobT, userID string, worker *workerT) bool {
```

Comment:
func (rt *HandleT) handleBackoff(job *jobsdb.JobT, userID string, worker *workerT) (shouldBackOff bool) {

Reply:
One thing to discuss is what happens if all 10k jobs are failed. Earlier and now too, the behavior would be similar though.

Earlier after 8 attempt we abort the job, but now till next n hours router will be blocked. But maybe we could at that time set retry_window to a small value and get them aborted?